src/components/slider/slider.tsx

```diff
@@ -1396,4 +1423,18 @@ export class Slider implements LabelableComponent, FormComponent, InteractiveCom
     const handleBounds = handle.getBoundingClientRect();
     return intersects(maxLabelBounds, handleBounds);
   }
+
+  /**
+   * Returns a string representing the localized label value based on groupSeparator prop being on or off.
+   *
+   * @param value
+   */
+  private determineGroupSeparator = (value): string => {
+    const lang = this.globalAttributes["lang"] || document.documentElement.lang || "en";
```

Comment:
I wonder instead of `document.documentElement.lang` we should do `this.el.closes(["lang"]).lang`? What do you think @jcfranco @Elijbet ? Right now, it doesn't honor parent elements lang.

I'm ok with enhancing this in globalAttributes helper if we can do that instead. Maybe we can enhance globalAttributes to do closest()?

Reply:
This is a compromise right now to avoid querying the DOM between renders. We can doc this limitation, create an issue for adding `lang` lookup and wait for user feedback since users would be expect that to work. cc @geospatialem @benelan